src/fmu/dataio/_utils.py

```diff
@@ -155,11 +155,6 @@ def nested_parameters_dict(
             unique_keys.append(key)
             nested_dict[key] = value
 
-    if len(unique_keys) > len(set(unique_keys)):
-        raise ValueError(
-            "Keys in parameters dictionary are not unique after removal of namespace"
-        )
-
```

Comment:
I think when we look in semeio: https://github.com/equinor/semeio/blob/24bc00bb1755ab420bc4b8e1f1be57c7f4096086/semeio/jobs/design_kw/design_kw.py, this is implemented there _with some given conditions_ (one is that it is checked for DESIGN_KW only). In that sense this can be removed here